netbox/dcim/tables/devices.py

```diff
@@ -597,6 +598,43 @@ def value_tagged_vlans(self, value):
         return ",".join([str(obj) for obj in value.all()])
 
 
+class MACAddressTable(NetBoxTable):
+    mac_address = tables.Column(
+        verbose_name=_('MAC Address'),
+        linkify=True
+    )
+    assigned_object = tables.Column(
+        linkify=True,
+        orderable=False,
+        verbose_name=_('Interface')
+    )
+    is_primary = columns.BooleanColumn(
+        verbose_name=_('Primary MAC'),
+        false_mark=None
+    )
+    assigned_device = tables.Column(
+        accessor='assigned_object__device',
+        verbose_name=_('Device'),
+        linkify=True
+    )
+    assigned_vm = tables.Column(
+        accessor='assigned_object__virtual_machine',
+        verbose_name=_('Virtual Machine'),
+        linkify=True
+    )
```

Comment:
Agreed, I'd rather follow the IPAddressTable pattern than break it here for no good reason. I won't add an `assigned` column though because it doesn't seem very valuable for MACAddresses (they should always be assigned).